python/dgl/data/rdf.py

```diff
@@ -142,7 +142,12 @@ def process_raw_tuples(self, raw_tuples):
         dst = []
         ntid = []
         etid = []
-        for i, (sbj, pred, obj) in enumerate(raw_tuples):
+        sorted_tuples = []
+        for t in raw_tuples:
+            sorted_tuples.append(t)
+        sorted_tuples.sort()
+
+        for i, (sbj, pred, obj) in enumerate(sorted_tuples):
```

Comment:
That's interesting. This told us that we'd better make our code deterministic if graph structure is generated from some raw input data because users might develop transductive models that are sensitive to the actual order.

Reply:
Yes, this is the case when we separate training and inference.
This should be carefully handled, otherwise, the model is meaningless.